### PR TITLE
Pickup HUD Element

### PR DESCRIPTION
**Pull request by lebroomer**
* Refactoring of cl_hudpickup
* New HUDElement for the pickups

### CODE REVIEW

**Comment by TimGoll** [comment]
Ah, and the PureSkin variant isn't shown in the HUD switcher. But since I'm going to make the design of it I can change it

File: gamemodes/terrortown/gamemode/client/cl_hudpickup.lua [hunk 1]
```diff
@@ -24,26 +22,17 @@ function GM:HUDWeaponPickedUp(wep)
...
-	pickup.width = w
-
-	if self.PickupHistoryLast >= pickup.time then
-		pickup.time = self.PickupHistoryLast + 0.05
+	if PICKUP.last >= pickup.time then
+		pickup.time = PICKUP.last + 0.05
 	end
 
-	table.insert(self.PickupHistory, pickup)
+	local index = table.insert(PICKUP.items, pickup)
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/client/cl_hudpickup.lua]: What is the index saved for?

File: gamemodes/terrortown/gamemode/client/cl_hudpickup.lua [hunk 2]
```diff
@@ -57,26 +46,17 @@ function GM:HUDItemPickedUp(itemname)
...
-		pickup.time = self.PickupHistoryLast + 0.05
+	if PICKUP.last >= pickup.time then
+		pickup.time = PICKUP.last + 0.05
 	end
 
-	table.insert(self.PickupHistory, pickup)
+	table.insert(PICKUP.items, pickup)
 
-	self.PickupHistoryLast = pickup.time
+	PICKUP.last = pickup.time
```
Comment (saibotk) [inline gamemodes/terrortown/gamemode/client/cl_hudpickup.lua]: Correct me if i am wrong, but isn't this the same code as in HUDWeaponPickup? If so i would suggest putting that in one function, to reduce redundancy, if there is no good reason for it to be standalone.

**Comment by lebroomer** [comment]
The PR wasn't open for a review, thatswhy I marked it as draft. So of course the code quality isn't perfect and will be overhauled before the merge ^^

**Comment by saibotk** [comment]
> The PR wasn't open for a review, thatswhy I marked it as draft. So of course the code quality isn't perfect and will be overhauled before the merge ^^

But you marked me to review, so i did, next time mark this as a WIP PR in Github.

**Comment by lebroomer** [comment]
It was, @Alf21 removed the WIP label ^^

**Comment by TimGoll** [comment]
I can confirm, there was WIP label. But there's no mention of someone removing it. Strange.

**Comment by lebroomer** [comment]
@Alf21 Alf21 marked this pull request as ready for review yesterday

**Comment by Alf21** [comment]
Got a GitHub Notification to review -> I reviewed.

**Comment by lebroomer** [comment]
Maybe this automatically changed the PR from "draft" to "ready to review". Who knows..

**Pull request merged**